bindings/nodejs/src/structures/oic-client.cc

```
@@ -0,0 +1,128 @@
+    Nan::Set(jsResource, Nan::New("interfaces").ToLocalChecked(),
+        jsStringArrayFromStrSliceVector(&(resource->interfaces)));
+    Nan::Set(jsResource, Nan::New("is_observing").ToLocalChecked(),
+        Nan::New(resource->is_observing));
+    Nan::Set(jsResource, Nan::New("observable").ToLocalChecked(),
+        Nan::New(resource->observable));
+    Nan::Set(jsResource, Nan::New("secure").ToLocalChecked(),
+        Nan::New(resource->secure));
+    Nan::Set(jsResource, Nan::New("types").ToLocalChecked(),
+        jsStringArrayFromStrSliceVector(&(resource->types)));
+
+    return jsResource;
+}
+
+const char *SolOicClient::jsClassName() {
+    return "SolOicClient";
+}
+
+void SolOicClient::ref(void *data) {}
+
+void SolOicClient::unref(void *data) {
+    sol_oic_client_del((sol_oic_client *)data);
+}
+
+OicCallbackData::OicCallbackData(): jsClient(0), callback(0),
+    hijackRefWasSuccessful(false) {}
+OicCallbackData::~OicCallbackData() {
+    if (jsClient) {
+        jsClient->Reset();
+        delete jsClient;
+    }
+    delete callback;
+    if (hijackRefWasSuccessful) {
+        hijack_unref();
+    }
+}
+
+bool OicCallbackData::init(Local<Object> _jsClient,
+    Local<Function> jsCallback) {
+    callback = new Nan::Callback(jsCallback);
+    if (!callback) {
+        Nan::ThrowError("OicCallbackData: Failed to allocate callback");
+        return false;
+    }
+    jsClient = new Nan::Persistent<Object>(_jsClient);
+    if (!jsClient) {
+        delete callback;
+        Nan::ThrowError("OicCallbackData: Failed to allocate client");
+        return false;
+    }
+
+    hijackRefWasSuccessful = hijack_ref();
+    return hijackRefWasSuccessful;
+}
+
+OicCallbackData *OicCallbackData::New(Local<Object> jsClient,
+    Local<Function> jsCallback) {
+    OicCallbackData *data = new OicCallbackData;
+    if (!data) {
+        Nan::ThrowError("Failed to allocate OicCallbackData");
+    } else if (!data->init(jsClient, jsCallback)) {
```

is it expected the failure of the init call ? or should throw an exception also here ?